azure-iot-device/azure/iot/device/iothub/sync_clients.py

```diff
@@ -323,6 +323,47 @@ def receive_twin_desired_properties_patch(self, block=True, timeout=None):
         logger.info("twin patch received")
         return patch
 
+    def get_blob_shared_access_signature(self, blob_name):
+        """Call up to the IoT Hub Endpoint over HTTP to get information on
+        the storage blob for uploads.
+        """
+        # TODO: Check that the HTTP Pipeline has been set properly.
+        if not self._upload_pipeline:
+            # raise error
+            raise exceptions.ClientError(
+                "No Upload Pipeline Initialized. get_blob_shared_access_signature cannot be called without an Upload Pipeline set."
+            )
+        else:
+            callback = EventedCallback(return_arg_name="storage_info")
+            self._upload_pipeline.get_blob_shared_access_signature(
+                blob_name=blob_name, callback=callback
+            )
+            storage_info = handle_result(callback)
+            logger.info("Successfully retrieved storage_info")
+            return storage_info
+
+    def notify_blob_upload_status(
+        self, correlation_id, upload_response, status_code, status_description
```

Comment:
what is a correlation id and where does the caller get it?

Reply:
correlation_id is used by IoT Hub to correlate the upload status of each blob you are communicating with. I can add documentation specifying this. 

---
In reply to: [352787749](https://github.com/Azure/azure-iot-sdk-python/pull/403#discussion_r352787749) [](ancestors = 352787749)